dask/array/linalg.py

```diff
@@ -328,3 +328,117 @@ def svd(a, name=None):
+        if i == 0:
+            dsk[(name_lu, i, i)] = (scipy.linalg.lu, (a.name, i, i))
+            for j in range(i + 1, hdim):
+                # sweep to horizontal
+                dsk[(name_lu, i, j)] = (_lu_horizontal, (a.name, i, j), (name_l_internal, i, i))
+            for k in range(i + 1, vdim):
+                # sweep to vertical
+                dsk[(name_lu, k, i)] = (_lu_vertical, (a.name, k, i), (name_u_internal, i, i))
+        else:
+            dsk[(name_lu, i, i)] = (_lu_diag, (a.name, i, i), (name_l_internal, i, i - 1),
+                                    (name_u_internal, i - 1, i))
+
+            for j in range(i + 1, hdim):
+                # sweep to horizontal
+                dsk[(name_lu, i, j)] = (_lu_horizontal_diag, (a.name, i, j),
+                                                             (name_l_internal, i, 1),
+                                                             (name_u_internal, i - 1, j),
+                                                             (name_l_internal, i, i))
+            for k in range(i + 1, vdim):
+                # sweep to vertical
+                dsk[(name_lu, k, i)] = (_lu_vertical_diag, (a.name, k, i),
+                                                           (name_l_internal, k, 1),
+                                                           (name_u_internal, 1, i)
+                                                           (name_u_internal, i, i))
+
+        for j in range(i + 1, hdim):
+            # sweep to horizontal
+            dsk[(name_lu, i, j)] = (_lu_horizontal, (a.name, i, j), (name_l_internal, i, i))
+        for k in range(i + 1, vdim):
+            # sweep to vertical
+            dsk[(name_lu, k, i)] = (_lu_vertical, (a.name, k, i), (name_u_internal, i, i))
+
+    for i in range(min(vdim, hdim)):
+        for j in range(min(vdim, hdim)):
+            if i == j:
+                dsk[(name_l_internal, i, j)] = (operator.getitem, (name_lu, i, j), 1)
+                dsk[(name_u_internal, i, j)] = (operator.getitem, (name_lu, i, j), 2)
+
+                dsk[(name_p, i, j)] = (operator.getitem, (name_lu, i, j), 0)
+
+            elif i > j:
+                # ToDo: remove _dummy_identity
+                dsk[(name_p, i, j)] = (np.zeros, (a.chunks[0][i], a.chunks[1][j]))
+                dsk[(name_l_internal, i, j)] = (_dummy_identity, (name_lu, i, j))
+                dsk[(name_u_internal, i, j)] = (np.zeros, (a.chunks[0][i], a.chunks[1][j]))
+            else:
+                dsk[(name_p, i, j)] = (np.zeros, (a.chunks[0][i], a.chunks[1][j]))
+                dsk[(name_l_internal, i, j)] = (np.zeros, (a.chunks[0][i], a.chunks[1][j]))
+                dsk[(name_u_internal, i, j)] = (_dummy_identity, (name_lu, i, j))
+
+            dsk[(name_l, i, j)] = (_dummy_identity, (name_l_internal, i, j))
+            dsk[(name_u, i, j)] = (_dummy_identity, (name_u_internal, i, j))
+
+    dsk.update(a.dask)
+
+    dsk_p = dsk.copy()
```

Comment:
It's probably not necessary, so far all operations within dask are pure.  However, it's also very cheap; we copy large dicts all the time.

```python
In [1]: import dask.array as da
In [2]: x = da.random.random((10000, 10000), chunks=(100, 100))

In [3]: %time y = ((x + 1)**2).sum(axis=0).mean()
CPU times: user 473 ms, sys: 4.31 ms, total: 477 ms
Wall time: 477 ms

In [4]: %time _ = y.dask.copy()
CPU times: user 6.56 ms, sys: 0 ns, total: 6.56 ms
Wall time: 6 ms

In [5]: len(y.dask)
Out[5]: 40605
```